mountpoint-s3-fs/src/metrics.rs

```diff
@@ -115,15 +166,39 @@ impl MetricsSink {
     }
 
     /// Publish all this sink's metrics to `tracing` log messages
+    /// and send metrics to OTLP if enabled
     fn publish(&self) {
         // Collect the output lines so we can sort them to make reading easier
         let mut metrics = vec![];
 
         for mut entry in self.metrics.iter_mut() {
             let (key, metric) = entry.pair_mut();
-            let Some(metric) = metric.fmt_and_reset() else {
+
+            #[cfg(feature = "otlp_integration")]
+            // Get both the value and string representation of the metric (this also resets the metric)
```

Comment:
Perhaps this should be split out into two functions, one with the feature, and one without. I think that would increase the readability here